### PR TITLE
Add hyper-v metrics

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
Add hyper v metrics -- charts for 
- VM Memory Pressure
- VM Assigned Memory
- VM Guest Visible Memory
- VM Physical Pages Allocated
- VM Physical Pages Not Allocated from Preferred NUMA Node
- Virtual Machines Health Status
- Root Partition Device Space Pages
- Root Partition GPA Space Pages
- Root Partition GPA Space Modifications
- Root Partition Attached Devices
- Root Partition Deposited Pages
- Root Partition Illegal DMA Requests
- Root Partition Illegal Interrupt Requests
- Root Partition Throttled Interrupts
- Root Partition Flushes of I/O TLBs
- Root Partition Address Spaces in Virtual TLB
- Root Partition Pages Used by Virtual TLB
- Root Partition Flushes of Entire Virtual TLB
- VM Storage Device IOPS
- VM Storage Device IO
- VM Storage Device Errors
- Virtual Switch Traffic
- Virtual Switch Packets
- Virtual Switch Directed Packets
- Virtual Switch Broadcast Packets
- Virtual Switch Multicast Packets
- Virtual Switch Dropped Packets
- Virtual Switch Extensions Dropped Packets
- Virtual Switch Flooded Packets
- Virtual Switch Learned MAC Addresses
- Virtual Switch Purged MAC Addresses
- VM Interface Packets Dropped
- VM CPU Usage

### CODE REVIEW

File: src/collectors/windows.plugin/windows_plugin.c [hunk 1]
```diff
@@ -25,6 +25,7 @@ static struct proc_module {
     {.name = "PerflibStorage",      .dim = "PerflibStorage",       .enabled = CONFIG_BOOLEAN_YES, .func = do_PerflibStorage},
     {.name = "PerflibNetwork",      .dim = "PerflibNetwork",       .enabled = CONFIG_BOOLEAN_YES, .func = do_PerflibNetwork},
     {.name = "PerflibObjects",      .dim = "PerflibObjects",       .enabled = CONFIG_BOOLEAN_YES, .func = do_PerflibObjects},
+    {.name = "PerflibHyperV",       .dim = "PerflibHyperV",      .enabled = CONFIG_BOOLEAN_YES, .func = do_PerflibHyperV},
```
Comment (thiagoftsm) [inline src/collectors/windows.plugin/windows_plugin.c:28]: A minor here, please, take a look in the alignment of `.enable`.

File: src/collectors/windows.plugin/perflib-hyperv.c [hunk 2]
```diff
@@ -0,0 +1,1553 @@
...
+#define _COMMON_PLUGIN_NAME "windows.plugin"
+#define _COMMON_PLUGIN_MODULE_NAME "PerflibHyperV"
+#include "../common-contexts/common-contexts.h"
+
+#define HYPERV  "hyperv"
+
+
+long chart_priority = NETDATA_CHART_PRIO_WINDOWS_HYPERV;
+
+void get_and_sanitize_instance_value(PERF_DATA_BLOCK *pDataBlock,  PERF_OBJECT_TYPE *pObjectType, PERF_INSTANCE_DEFINITION *pi, char *buffer, size_t buffer_size)
```
Comment (thiagoftsm) [inline src/collectors/windows.plugin/perflib-hyperv.c:15]: I loved this function, I would suggest to move it to `windows_plugin.h` as static inline, or to windows.plugin main file.

File: src/collectors/windows.plugin/perflib-hyperv.c [hunk 3]
```diff
@@ -0,0 +1,1553 @@
...
+        strncpyz(buffer, "[unknown]", buffer_size - 1);
+        return;
+    }
+    rrdlabels_sanitize_value(buffer, wstr, buffer_size);
+}
+
+#define DICT_PERF_OPTION (DICT_OPTION_DONT_OVERWRITE_VALUE | DICT_OPTION_FIXED_SIZE)
+
+#define DEFINE_RD(counter_name) RRDDIM *rd_##counter_name
+
```
Comment (thiagoftsm) [inline src/collectors/windows.plugin/perflib-hyperv.c:28]: We could also do this with the following macro, appending a prefix like `ND_PERFLIB`.

File: src/collectors/windows.plugin/perflib-hyperv.c [hunk 4]
```diff
@@ -0,0 +1,1553 @@
...
+        GET_INSTANCE_COUNTER(ExtensionsDroppedPacketsOutgoingSec);
+        GET_INSTANCE_COUNTER(ExtensionsDroppedPacketsIncomingSec);
+
+        GET_INSTANCE_COUNTER(PacketsFlooded);
+
+        GET_INSTANCE_COUNTER(LearnedMacAddresses);
+
+        GET_INSTANCE_COUNTER(PurgedMacAddresses);
+
+            // Create charts
```
Comment (thiagoftsm) [inline src/collectors/windows.plugin/perflib-hyperv.c:1052]: these lines have a different tabulation.

File: src/collectors/windows.plugin/perflib-hyperv.c (hunk 3 above)
Comment (stelfrag) [reply src/collectors/windows.plugin/perflib-hyperv.c:28]: What do you mean?

Comment (thiagoftsm) [reply src/collectors/windows.plugin/perflib-hyperv.c:17]: I thought name `DEFINE_RD` too generic.  So my suggestion was do rename it to `ND_DEFINE_RD`, for example.

**Review by thiagoftsm — APPROVED** [review]
PR is working as expecting. LGTM!

**Review by thiagoftsm — APPROVED** [review]
I retest the PR  and it continue working after last commits.

**Comment by ktsaou** [comment]
@stelfrag can we merge this please?

**Pull request merged**